resources/js/views/settings/roomTypes/View.vue

```
@@ -0,0 +1,186 @@
+          <b-form-input id='color' type='text' v-model='model.color' :state='fieldState("color")' :disabled='isBusy || viewOnly'></b-form-input>
+
+          <template slot='invalid-feedback'><div v-html="fieldError('color')"></div></template>
+        </b-form-group>
+
+        <b-form-group
+          label-cols-sm='4'
+          :label="$t('settings.roomTypes.preview')"
+        >
+          <div class="roomicon" :style="{ 'background-color': model.color}">{{ model.short }}</div>
+        </b-form-group>
+
+        <hr>
+        <b-row class='my-1 float-right'>
+          <b-col sm='12'>
+            <b-button
+              :disabled='isBusy'
+              variant='secondary'
+              @click="$router.push({ name: 'settings.room_types' })">
+              <i class='fas fa-arrow-left'></i> {{ $t('app.back') }}
+            </b-button>
+            <b-button
+              :disabled='isBusy'
+              variant='success'
+              type='submit'
+              class='ml-1'
+              v-if='!viewOnly'>
+              <i class='fas fa-save'></i> {{ $t('app.save') }}
+            </b-button>
+          </b-col>
+        </b-row>
+      </b-container>
+    </b-form>
+
+  </div>
+</template>
+
+<script>
+import Base from '../../../api/base';
+import FieldErrors from '../../../mixins/FieldErrors';
+import { mapGetters } from 'vuex';
+import env from '../../../env';
+import VSwatches from 'vue-swatches';
+import 'vue-swatches/dist/vue-swatches.css';
+
+export default {
+  mixins: [FieldErrors],
+  components: {
+    VSwatches
+  },
+  props: {
+    id: {
+      type: [String, Number],
+      required: true
+    },
+
+    viewOnly: {
+      type: Boolean,
+      required: true
+    },
+
+    modalStatic: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      settings: 'session/settings'
+    })
+
+  },
+
+  data () {
+    return {
+      isBusy: false,
+      errors: {},
+      staleError: {},
```

Stale errors are not handled, but they should be handled!